redfish/provider/common.go

```diff
@@ -265,7 +265,7 @@ func (p powerOperator) PowerOperation(resetType string, maximumWaitTime int64, c
 }
 
 // checkServerStatus checks iDRAC server status after provided interval until the provided timeout time
-func checkServerStatus(ctx context.Context, endpoint string, interval int, timeout int) error {
+func checkServerStatus(ctx context.Context, service *gofish.Service, endpoint string, timeoutInterval map[string]int) error {
```

Comment:
I am wary of using arbitrary maps where properly defined structs can be used instead.
How about this. Can we define a struct that looks something like 
```go
type ServerStatusChecker struct {
    Service *gofish.Service
    Endpoint string
    Interval int
    Timeout int
}
```
And then create a method like
```go
func (s *ServerStatusChecker) Check(ctx context.Context) error {
...
}
```
?

Then it can be called like
```go
checker := ServerStatusChecker{
    Service: service,
    Endpoint: (*params.rserver)[0].Endpoint.ValueString(),
    Interval: defaultCheckInterval,
    Timeout: defaultCheckTimeout,
}
err = checker.Check(ctx)
```

Reply:
done